src/lib/plugins/useItem.js

```diff
@@ -27,10 +40,24 @@ module.exports.server = (serv, { version }) => {
         }
       }
     } else {
-      serv.onItemPlace('spawn_egg', ({ heldItem, player, placedPosition }) => {
-        serv.spawnMob(getEntID(heldItem.nbt.value.EntityTag.value.id.value), player.world, placedPosition)
-        return { id: -1, data: 0 }
-      })
+      if (version === '1.12.2') {
+        serv.onItemPlace('spawn_egg', ({ item, player, placedPosition }) => {
+          serv.spawnMob(getEntIDmc(item.nbt.value.EntityTag.value.id.value), player.world, placedPosition)
+          return { id: -1, data: 0 }
+        })
+      } else {
+        if (version === '1.8.8') {
+          serv.onItemPlace('spawn_egg', ({ item, player, placedPosition }) => {
+            serv.spawnMob(item.metadata, player.world, placedPosition)
+            return { id: -1, data: 0 }
+          })
+        } else {
+          serv.onItemPlace('spawn_egg', ({ item, player, placedPosition }) => {
+            serv.spawnMob(getEntID(item.nbt.value.EntityTag.value.id.value), player.world, placedPosition)
```

Comment:
You could have only 1 getEntID function if you call it like:
```js
serv.spawnMob(getEntID('minecraft:' + item.nbt.value.EntityTag.value.id.value), player.world, placedPosition)

```

Reply:
Optimizing code wait a minute